app/gui/view/debug_scene/component-group/src/lib.rs

```diff
@@ -98,128 +101,132 @@ impl list_view::entry::ModelProvider<list_view::entry::Label> for MockEntries {
 
 
 
+// ==================================
+// === Component Group Controller ===
+// ==================================
+
+/// An example abstraction to arrange component groups inside the scroll area.
+#[derive(Debug, Clone, CloneRef)]
+struct ComponentGroupController {
+    component_groups: Rc<Vec<component_group::View>>,
+}
+
+impl ComponentGroupController {
+    fn init(
+        component_groups: &[component_group::View],
+        network: &frp::Network,
+        scroll_area: &ScrollArea,
+    ) {
+        Self { component_groups: Rc::new(component_groups.to_vec()) }
+            .init_inner(network, scroll_area)
+    }
+
+    fn init_inner(&self, network: &frp::Network, scroll_area: &ScrollArea) {
+        for (i, group) in self.component_groups.iter().enumerate() {
+            let this = self.clone_ref();
+            frp::extend! { network
+                eval group.size([group, this](size)
+                    group.set_position_y(-size.y / 2.0 - this.heights_sum(i))
+                );
+                is_scrolled <- scroll_area.scroll_position_y.map(f!([this] (s) *s > this.heights_sum(i)));
+                change_hdr_pos <- scroll_area.scroll_position_y.gate(&is_scrolled);
+                reset_hdr_pos <- is_scrolled.on_false();
+                eval change_hdr_pos([group, this](y) group.set_header_pos(*y - this.heights_sum(i)));
+                eval_ reset_hdr_pos(group.set_header_pos(0.0));
+            }
+        }
+    }
+
+    /// Return a sum of heights of first `n` component groups.
+    fn heights_sum(&self, n: usize) -> f32 {
+        self.component_groups.iter().take(n).map(|g| g.size.value().y).sum()
+    }
```

Comment:
Just an example of why these docs are a little bit bad here in general. "Return a sum of heights of first n component groups." - What does it mean "first" ? Are they counted from what side? I was reading the docs of different structures several times now, and I still don't see the answer anywhere, of how the layout works. Please, put some time in improving the docs in this PR in general.

Reply:
The docs are extended in all places.